examples/camera-app/linux/src/clusters/camera-avstream-mgmt/camera-av-stream-manager.cpp

```diff
@@ -386,42 +414,83 @@ Protocols::InteractionModel::Status CameraAVStreamManager::AudioStreamDeallocate
 Protocols::InteractionModel::Status CameraAVStreamManager::SnapshotStreamAllocate(const SnapshotStreamAllocateArgs & allocateArgs,
                                                                                   uint16_t & outStreamID)
 {
-    outStreamID = kInvalidStreamID;
+    outStreamID                                 = kInvalidStreamID;
+    bool isRequestSupportedByAnyAvailableStream = false;
 
-    for (SnapshotStream & stream : mCameraDeviceHAL->GetCameraHALInterface().GetAvailableSnapshotStreams())
+    // Check if allocation request can be supported
+    for (const auto & stream : mCameraDeviceHAL->GetCameraHALInterface().GetAvailableSnapshotStreams())
     {
         if (stream.IsCompatible(allocateArgs))
         {
-            outStreamID = stream.snapshotStreamParams.snapshotStreamID;
-            if (!stream.isAllocated)
-            {
-                stream.isAllocated = true;
+            isRequestSupportedByAnyAvailableStream = true;
+            break;
+        }
+    }
 
-                // Set the optional Watermark and OSD values that may have been provided.  This is the initial
-                // setting of these values, they may be subsequently modified. If the values have no value that
-                // is ok, the allocated stream will store as such and ignore.
-                stream.snapshotStreamParams.watermarkEnabled = allocateArgs.watermarkEnabled;
-                stream.snapshotStreamParams.OSDEnabled       = allocateArgs.OSDEnabled;
+    if (!isRequestSupportedByAnyAvailableStream)
+    {
+        return Status::DynamicConstraintError;
+    }
 
-                // Start the snapshot stream for serving.
-                mCameraDeviceHAL->GetCameraHALInterface().StartSnapshotStream(outStreamID);
-            }
-            else
-            {
-                ChipLogProgress(Camera, "Matching pre-allocated stream with ID: %d exists", outStreamID);
-            }
+    // Try to reuse an allocated stream.
+    for (auto & stream : GetCameraAVStreamMgmtServer()->GetAllocatedSnapshotStreams())
+    {
+        if (GetCameraAVStreamMgmtServer()->IsAllocatedSnapshotStreamReusable(stream, allocateArgs))
```

Comment:
So here: why isn't the API just "ask the cluster for a reusable stream ID"?  That would make it _much_ clearer when the API should be used (when you want to permit stream reuse at all) and would make it harder to mess up how it's used.